Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/UIExtensions.kt

```diff
@@ -13,3 +23,45 @@ fun NavController.doNavigate(direction: NavDirections) {
     currentDestination?.getAction(direction.actionId)
         ?.let { navigate(direction) }
 }
+
+/**
+ * Similar to [setupWithNavController],but it executes the passed action on item selection
+ * and shows [BottomNavigationView] on [HomeFragment], [ContactDiaryOverviewFragment] only
+ */
+fun BottomNavigationView.setupWithNavController2(
+    navController: NavController,
+    onItemSelected: () -> Unit
+) {
+    setupWithNavController(navController)
+    setOnNavigationItemSelectedListener { item ->
+        onItemSelected()
+        NavigationUI.onNavDestinationSelected(item, navController)
+    }
+    val weakBottomNavView = WeakReference(this)
```

Comment:
Why does this need to be a `WeakReference`?

Reply:
Actually, I tried here to have the same behaviour as the original setupWithNavController extension for nav-component.
I assume they do this as for the listener it might live after the view is gone

Reply:
Original code from the navigation components 
[
<img width="807" alt="Screenshot 2021-02-01 at 18 34 37" src="https://user-images.githubusercontent.com/25054729/106495597-32922580-64bc-11eb-907d-f9116cc593c0.png">
](url)